modules/websocket/websocket-destination.c

```diff
@@ -0,0 +1,223 @@
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->path);
+  self->path = g_strdup(path);
+}
+
+void websocket_dd_set_enable_ssl(LogDriver *destination, int flag) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  self->enable_ssl = flag;
+}
+
+void websocket_dd_set_cert(LogDriver *destination, gchar *path) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->cert);
+  self->cert = path == NULL ? NULL : g_strdup(path);
+}
+
+void websocket_dd_set_key(LogDriver *destination, gchar *path) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->key);
+  self->key = path == NULL ? NULL : g_strdup(path);
+}
+
+void websocket_dd_set_cacert(LogDriver *destination, gchar *path) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->cacert);
+  self->cacert = path == NULL ? NULL : g_strdup(path);
+}
+
+void websocket_dd_set_allow_self_signed(LogDriver *destination, int flag) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  self->allow_self_signed = flag;
+}
+
+void
+websocket_dd_set_template(LogDriver *destination, gchar *template)
+{
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  GlobalConfig* cfg = log_pipe_get_config(&destination->super);
+  self->template = log_template_new(cfg, NULL);
+  log_template_compile(self->template, template, NULL);
+}
+
+LogTemplateOptions *
+destination_dd_get_template_options(LogDriver *d)
+{
+  WebsocketDestDriver *self = (WebsocketDestDriver *)d;
+  return &self->template_options;
+}
+
+static void
+websocket_dd_free(LogPipe *destination)
+{
+  WebsocketDestDriver *self = (WebsocketDestDriver *) destination;
+  g_free(self->template);
+  g_free(self->address);
+  g_free(self->path);
+  g_free(self->protocol);
+  g_free(self->cert);
+  g_free(self->key);
+  g_free(self->cacert);
+  log_threaded_dest_driver_free(destination);
+}
+
+
+static void
+websocket_worker_thread_init(LogThrDestDriver *destination)
+{
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+
+  msg_debug("Worker thread started",
+            evt_tag_str("driver", self->super.super.super.id),
+            NULL);
+
+  websocket_client_create(self->protocol, self->address, self->port,
+    self->path, self->client_use_ssl_flag, self->cert, self->key, self->cacert);
+}
+
+
+static void
+websocket_dd_disconnect(LogThrDestDriver *destination)
+{
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  websocket_client_disconnect();
+}
+
+static worker_insert_result_t
+websocket_worker_insert(LogThrDestDriver *destination, LogMessage *msg)
+{
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  GString *result = g_string_new("");
+  log_template_format(self->template, msg, &self->template_options, LTZ_LOCAL, self->super.seq_num, NULL, result);
+  websocket_client_send_msg(result->str);
+  return WORKER_INSERT_RESULT_SUCCESS;;
+}
```

Comment:
Here you leak the result.
You allocate memory for it, use it, and return before release the allocated memory.